packages/amplify-graphql-transformer-core/src/migration/migrate.ts

```diff
@@ -0,0 +1,144 @@
+  await printer.info('If you confirm migration, you will receive no further prompts and all migration will be completed. The CLI will exit after migration and will not push your resources to give you time to review.');
+  await printer.warn('If you are using @auth, we strongly recommend spending extra time reviewing your auth rules after migration to ensure they\'re accurate');
+  return await prompter.confirmContinue("About to start migrating to V2 transformer, are you ready to continue?");
+}
+
+async function runMigration(schemas: SchemaDocument[], authMode: string): Promise<void> {
+  let fullSchema: string;
+  let schemaList: string[] = new Array(schemas.length);
+  let backupSchemaPromiseArray: Promise<void>[] = [];
+  schemas.forEach((doc, idx) => {
+    schemaList[idx] = doc.schema;
+    backupSchemaPromiseArray.push(backupSchema(doc.filePath));
+  });
+  for (let promise of backupSchemaPromiseArray) {
+    await promise;
+  }
+
+  fullSchema = schemaList.join('\n');
+  let fullSchemaNode = parse(fullSchema);
+  let writeNewPromiseArray: Promise<void>[] = [];
+  schemas.forEach(doc => {
+    const newSchema = migrateGraphQLSchema(doc.schema, authMode, fullSchemaNode);
+    writeNewPromiseArray.push(replaceFile(newSchema, doc.filePath));
+  });
+
+  for (let promise of writeNewPromiseArray) {
+    await promise;
+  }
+}
+
+export async function updateTransformerVersion(env: string): Promise<void> {
+  const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+  let envCLI = true;
+  let cliJSON: any;
+  try {
+    cliJSON = stateManager.getCLIJSON(projectPath, env);
+  }
+  catch (e) {
+    if (e.message.includes("File at path:") && e.message.includes("does not exist")) {
+      envCLI = false;
+      cliJSON = stateManager.getCLIJSON(projectPath);
+    }
+    else {
+      throw e;
+    }
+  }
+  if (!cliJSON.features) {
+    cliJSON.features = {};
+  }
+  if (!cliJSON.features.graphqltransformer) {
+    cliJSON.features.graphqltransformer = {}
+  }
+  cliJSON.features.graphqltransformer.useexperimentalpipelinedtransformer = true;
+
+  if (envCLI) {
+    stateManager.setCLIJSON(projectPath, cliJSON, env);
+  }
+  else {
+    stateManager.setCLIJSON(projectPath, cliJSON);
+  }
+  await FeatureFlags.reloadValues();
+}
+
+export async function migrateToV2Transformer(resourceDir: string, context: $TSContext): Promise<boolean> {
+  const { envName } = context.amplify.getEnvInfo();
+  const defaultAuth = await getDefaultAuthFromContext();
+  const authMode = cliToMigratorAuthMap.get(defaultAuth);
+  if (!authMode) {
+    throw Error(`Unidentified authorization mode for API found: ${defaultAuth}`);
+  }
+  const schemaFilePath = path.join(resourceDir, 'schema.graphql');
+  const schemaDirectoryPath = path.join(resourceDir, 'schema');
+  const schemaFileExists = await fs.existsSync(schemaFilePath);
+  const schemaDirectoryExists = await fs.existsSync(schemaDirectoryPath);
```

Comment:
No need to await synchronous functions.

Reply:
Yep I called the async first and forgot to delete the await